en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/defining-interceptors-in-an-open-api-definition.md

```diff
@@ -0,0 +1,245 @@
+.
+.
+```
+
+<!-- The content of the below warning is same as the info notice in the file
+deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/response-flow-interceptor.md -->
+!!! info
+    If you update the response body before reaching the client, make sure to add `response_body` to the `includes` section
+    of the Open API Specification. You may have a scenario like, what ever the response body from the backend, based on a header value
+    you define your own response body to be sent to the client. Even though you do not read the response body, the above inclusion
+    should be done to update the response body.
+
+## Description of Keys of Interceptor Open API Extension
+
+Following is the description of each key in the extension.
+
+| Open API Extension Key | Description                                                                                                            |
+|------------------------|------------------------------------------------------------------------------------------------------------------------|
+| scheme                 | HTTP protocol of interceptor service. One of `[http\|https]`                                                           |
+| host                   | Host of the interceptor service                                                                                        |
+| port                   | Port of the interceptor service                                                                                        |
+| includes               | Defines what should be included in the request body to interceptor service.                                            |
+| requestTimeout         | Optional. Timeout in seconds to connect to interceptor service.                                                        |
+| clusterTimeout         | Optional. Override the interceptor cluster timeout (in seconds) value, default to cluster timeout value in config.toml |
+
+### Inclusions in Request from Choreo Connect Router to Interceptor Service
+
+Based on the `includes` defined in the extension, the request from Choreo Connect Router to Interceptor Service includes only the specified information from **client request** or **backend response**.
+For example, with this configuration you can instruct the Choreo Connect Router to not include request body from
+the client and stop buffering the client request body and save the size of the request payload to interceptor service.
+In the same way you can use only required inclusions based on you requirement and tune the interception flow.
+
+Following are the possible values for the `includes` in the extension. More information on ***Invocation Context*** is can be found in
+[Interceptor Context and Invocation Context](http://localhost:8000/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/interceptor-context-and-invocation-context/#invocation-context)
```

Comment:
Please add a relative link here instead of localhost. You can use {{base_path}} instead.